rpcs3/Emu/Io/Dimensions.cpp

```diff
@@ -465,7 +501,10 @@ bool dimensions_toypad::move_figure(u8 pad, u8 index, u8 old_pad, u8 old_index)
 bool dimensions_toypad::create_blank_character(std::array<u8, 0x2D * 0x04>& buf, u16 id)
 {
 	random_uid(buf.data());
-	buf[7] = id & 0xFF;
+	buf[3] = id & 0xFF;
+
+	// LEGO Dimensions figures use NTAG213 tag types, and the UID for these is stored in
+	// bytes 0, 1, 2, 4, 5, 6 and 7 (out of 180 bytes)
 	std::array<u8, 7> uid = {buf[0], buf[1], buf[2], buf[4], buf[5], buf[6], buf[7]};
```

Comment:
This uid array is created unconditionally, but only used if id != 0.
So you can move it in there.